src/mlpack/methods/ann/layer/not_adapted/adaptive_max_pooling.hpp

```diff
@@ -13,44 +13,44 @@
 #define MLPACK_METHODS_ANN_LAYER_ADAPTIVE_MAX_POOLING_HPP
 
 #include <mlpack/prereqs.hpp>
-#include "layer_types.hpp"
+
+#include "layer.hpp"
+#include "max_pooling.hpp"
 
 namespace mlpack {
 namespace ann /** Artificial Neural Network. */ {
 
 /**
  * Implementation of the AdaptiveMaxPooling layer.
  *
- * @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
- *         arma::sp_mat or arma::cube).
- * @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
- *         arma::sp_mat or arma::cube).
+ * @tparam InputType The type of the layer's inputs. The layer automatically
+ *    cast inputs to this type (Default: arma::mat).
+ * @tparam OutputType The type of the layer's Outputs. The layer automatically
+ *    cast inputs to this type (Default: arma::mat).
  */
-template <
-    typename InputDataType = arma::mat,
-    typename OutputDataType = arma::mat
->
-class AdaptiveMaxPooling
+template <typename InputType = arma::mat, typename OutputType = arma::mat>
```

Comment:
Do we need the `OutputType` in this case?

Reply:
You are right, we don't.  For this particular class, I "sort of" adapted it, and then gave up because it wasn't needed for the examples in the examples repository.  This is true of some other layers too; so things in `not_adapted/` *might* be partially adapted already.  They will need a second pass, though.

Reply:
I see, thanks for the clarification.